it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/bigquery/BigQueryIOST.java

```diff
@@ -0,0 +1,595 @@
+        }
+        tableRow.set(
+            "data_" + (numColumns - 1),
+            encoder.encodeToString(Arrays.copyOfRange(input, curIdx, input.length)));
+      }
+      return tableRow;
+    }
+  }
+
+  /**
+   * Generates and returns a list of LoadPeriod instances representing periods of load increase
+   * based on the specified load increase array and total duration in minutes.
+   *
+   * @param minutesTotal The total duration in minutes for which the load periods are generated.
+   * @return A list of LoadPeriod instances defining periods of load increase.
+   */
+  private List<LoadPeriod> getLoadPeriods(int minutesTotal, int[] loadIncreaseArray) {
+
+    List<LoadPeriod> loadPeriods = new ArrayList<>();
+    long periodDurationMillis =
+        Duration.ofMinutes(minutesTotal / loadIncreaseArray.length).toMillis();
+    long startTimeMillis = 0;
+
+    for (int loadIncreaseMultiplier : loadIncreaseArray) {
+      long endTimeMillis = startTimeMillis + periodDurationMillis;
+      loadPeriods.add(new LoadPeriod(loadIncreaseMultiplier, startTimeMillis, endTimeMillis));
+
+      startTimeMillis = endTimeMillis;
+    }
+    return loadPeriods;
+  }
+
+  private enum WriteFormat {
+    AVRO,
+    JSON
+  }
+
+  /** Options for Bigquery IO stress test. */
+  static class Configuration extends SyntheticSourceOptions {
+
+    /**
+     * Number of columns of each record. The column size is equally distributed as
+     * valueSizeBytes/numColumns.
+     */
+    @JsonProperty public int numColumns = 1;
+
+    /** Pipeline timeout in minutes. Must be a positive value. */
+    @JsonProperty public int pipelineTimeout = 20;
+
+    /** Runner specified to run the pipeline. */
+    @JsonProperty public String runner = "DirectRunner";
+
+    /** Number of workers for the pipeline. */
+    @JsonProperty public int numWorkers = 20;
+
+    /** Maximum number of workers for the pipeline. */
+    @JsonProperty public int maxNumWorkers = 100;
+
+    /** BigQuery write method: DEFAULT/FILE_LOADS/STREAMING_INSERTS/STORAGE_WRITE_API. */
+    @JsonProperty public String writeMethod = "DEFAULT";
+
+    /** BigQuery write format: AVRO/JSON. */
+    @JsonProperty public String writeFormat = "AVRO";
+
+    /**
+     * Rate of generated elements sent to the source table. Will run with a minimum of 1k rows per
+     * second.
+     */
+    @JsonProperty public int rowsPerSecond = DEFAULT_ROWS_PER_SECOND;
+
+    /** Rows will be generated for this many minutes. */
+    @JsonProperty public int minutes = 15;
+
+    /**
+     * Determines the destination for exporting metrics. If set to true, metrics will be exported to
+     * InfluxDB and displayed using Grafana. If set to false, metrics will be exported to BigQuery
+     * and displayed with Looker Studio.
+     */
+    @JsonProperty public boolean exportMetricsToInfluxDB = false;
+
+    /** InfluxDB measurement to publish results to. * */
+    @JsonProperty public String influxMeasurement;
```

Comment:
(A note for future dashboard) Would the measurement be unique for each stress test / each IO ?

Pasted an example metrics fyi
```
{
  "TotalStreamingDataProcessed": 0.0,
  "BillableShuffleDataProcessed": 0.0,
  "EstimatedCost": 0.06965586137166667,
  "AvgInputThroughputBytesPerSec": 6.712440411666667E7,
  "ElapsedTime": 1811.0,
  "MaxCpuUtilization": 0.7093815630496711,
  "AvgCpuUtilization": 0.6400222870148885,
  "AvgInputThroughputElementsPerSec": 66132.418,
  "TotalPdUsage": 405457.0,
  "TotalGpuTime": 0.0,
  "TotalSsdUsage": 0.0,
  "MaxInputThroughputElementsPerSec": 71111.15,
  "TotalDcuUsage": 0.0,
  "TotalVcpuTime": 3243.0,
  "TotalShuffleDataProcessed": 0.0,
  "EstimatedDataProcessedGB": 101.5,
  "TotalMemoryUsage": 1.3286034E7,
  "MaxInputThroughputBytesPerSec": 7.217781435E7
}
``` 

As can be seen, the field name obtained from getMetrics is not are the same for all pipelines. If we start to publish to influxDB, we need to consider the naming of measurement field to distinguish different test settings.

Even better is to use influxDB tags to distinguish them, however currently it is not supported by IOITMetrics. 

https://docs.influxdata.com/influxdb/v1/concepts/glossary/#measurement

Reply:
Fixed using suffixes, so the measurement should be unique for each stress test case.

It's a good idea to use tags, we'll look into that in the future.